astropy/time/formats.py

```diff
@@ -577,17 +588,18 @@ class TimeFromEpoch(TimeNumeric):
     or days).
     """
 
-    def __init__(self, val1, val2, scale, precision,
-                 in_subfmt, out_subfmt, from_jd=False):
-        self.scale = scale
-        # Initialize the reference epoch (a single time defined in subclasses)
-        epoch = Time(self.epoch_val, self.epoch_val2, scale=self.epoch_scale,
-                     format=self.epoch_format)
-        self.epoch = epoch
+    @classproperty(lazy=True)
```

Comment:
Yes, good way to move forward with this - even if obviously it would be nice to solve it eventually!

And nice to see that the whole `__init__` could now go!